hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -331,3 +337,119 @@ func (a OpenStack) DeleteCredentials(ctx context.Context, c client.Client, hclus
 	// }
 	return nil
 }
+
+func machineNetworksToStrings(machineNetworks []hyperv1.MachineNetworkEntry) []string {
+	var machineNetworksStr []string
+	for _, machineNetwork := range machineNetworks {
+		machineNetworksStr = append(machineNetworksStr, machineNetwork.CIDR.String())
+	}
+	return machineNetworksStr
+}
+
+func defaultWorkerSecurityGroupRules(machineCIDRs []string) []capo.SecurityGroupRuleSpec {
+	ingressRules := []capo.SecurityGroupRuleSpec{}
+
+	// Rules for worker security group with the remote IP prefix set to the machine CIDRs
+	for _, machineCIDR := range machineCIDRs {
+		machineCIDRInboundRules := []capo.SecurityGroupRuleSpec{
+			{
+				Name:     "esp-ingress",
+				Protocol: k8sutilspointer.String("esp"),
+			},
+			{
+				Name:     "icmp-ingress",
```

Comment:
shouldn't ping be allowed from everywhere?

Reply:
AWS rules in HCP:

API 	 6443 	     tcp machine CIDRs for private link endpoint
API 	 6443 	     udp machine CIDRs for private link endpoint
Geneve 	 6081 udp
HTTPS 	 443 	     tcp machine CIDRs
HTTPS 	 443 	     udp machine CIDRs
ICMP                 machine CIDRs
IKE 	 500 udp
IKE NAT	 4500 udp
Internal 9000 9999   tcp
Internal 9000 9999   udp
Kubelet  10250       tcp
Nodeport 30000 32767 tcp
Nodeport 30000 32767 udp
SSH 	 22 	     tcp machine CIDRs
VXLAN 	 4789 udp

Honestly I don't think allowing ICMP from anywhere is a good idea but we can relax it later eventually.